.github/workflows/docker-build-and-publish.yaml

```diff
@@ -19,37 +19,50 @@ on:
         required: false
         default: ""
         type: string
-      image-artifact-name:
-        description: "Name of the artifact that contains the Docker image.tar file to push, see https://github.com/actions/upload-artifact (Default is 'image-artifact')."
+      image-repository:
+        description: "Repository of Docker image."
         required: false
-        default: "image-artifact"
+        default: ""
         type: string
       image-name:
-        description: "Name of Docker image (Default is the repository name)."
+        description: "Name of Docker image."
         required: false
         default: "${{ github.event.repository.name }}"
         type: string
+      image-tag:
+        description: "Tag of Docker image."
+        required: false
+        default: "pipeline-${{ github.run_id }}-git-${GITHUB_SHA::8}"
+        type: string
       ref:
-        description: "Ref to checkout"
+        description: "Ref name to checkout"
         required: false
         default: ""
         type: string
+      retention-days:
+        description: "Number of days the image artifact should be stored on GitHub."
+        required: false
+        default: 1
```

Comment:
is this a sensible default? the `upload-artifact` action uses 90 days by default

Reply:
I guess we only need it temporarily until it is published to the registry, right?